src/ExtractTileOperations.cpp

```diff
@@ -143,6 +164,138 @@ Tile<3> get_3d_tile_index(const Expr &e) {
     return {true, base, {x_stride, 0, r_stride}, {x_tile, y_tile, r_tile}};
 }
 
+Tile<3> get_3d_rhs_tile_index(const Expr &e, int element_width) {
```

Comment:
Perhaps not necessary for this PR, but we really should express these as pattern-based rules, they would be far easier to parse and debug